### PR TITLE
Make Server class more extensible

### PR DESCRIPTION
**Pull request by abner**
It fixes issue #49. 

- Added sessionDisabled attribute to allow disable session (for allow create serveless servers);
- Changed methods loadFileConfig, loadConfig and setupSwagger visibility to protected;
- Added method configureVertxOptions to allow customize VertxOptions used to create the Vertx object;
- Added test for sessionDisabled configuration

### CODE REVIEW

File: redpipe-engine/src/main/java/net/redpipe/engine/core/Server.java [hunk 1]
```diff
@@ -233,59 +240,73 @@ private Completable startVertx(VertxResteasyDeployment deployment)
...
-			        	if(previousValue != null) {
-			        		io.netty.handler.codec.http.cookie.Cookie nettyCookie = ClientCookieDecoder.LAX.decode(previousValue);
-			        		Cookie newCookie = Cookie.newInstance(io.vertx.ext.web.Cookie.cookie(nettyCookie));
-			        		context.addCookie(newCookie);
-			        	}
-			        }
-			      }
+		AppGlobals globals = AppGlobals.get();
+
+		boolean sessionDisabled = new Boolean(true).equals(globals.getConfig().getBoolean("sessionDisabled"));
```
Comment (FroMage) [inline redpipe-engine/src/main/java/net/redpipe/engine/core/Server.java]: This should really be `Boolean.TRUE`

**Comment by FroMage** [comment]
LGTM aside from the minor remark. If you can fix it I'll merge this ASAP. Thanks!

File: redpipe-engine/src/main/java/net/redpipe/engine/core/Server.java (hunk 1 above)
Comment (abner) [reply redpipe-engine/src/main/java/net/redpipe/engine/core/Server.java]: just fixed it 8a2f8ddd7394ce0e12d6490197c3daa59d240b94

**Comment by abner** [comment]
> LGTM aside from the minor remark. If you can fix it I'll merge this ASAP. Thanks!

Ok, i just got it fixed at commit 8a2f8ddd7394ce0e12d6490197c3daa59d240b94

**Comment by abner** [comment]
Hi. There is an idea when this PR would be accepted?

**Comment by FroMage** [comment]
Right now. Thanks a lot!

**Pull request merged**